app/javascript/mastodon/components/status_quoted.tsx

```diff
@@ -158,7 +158,11 @@ interface StatusQuoteManagerProps {
  */
 
 export const StatusQuoteManager = (props: StatusQuoteManagerProps) => {
-  const status = useAppSelector((state) => state.statuses.get(props.id));
+  const status = useAppSelector((state) => {
+    const status = state.statuses.get(props.id);
+    const reblogId = status?.get('reblog') as string | undefined;
+    return reblogId ? state.statuses.get(reblogId) : status;
```

Comment:
Would this be easier to read with a single call to `statuses.get()`?

```suggestion
    const reblogId = status?.get('reblog') as string | undefined;
    return state.statuses.get(reblogId ?? props.id);
```

Reply:
Where is the `status` coming from in this snippet?